### PR TITLE
Flyte address resolution

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Add a quick lookup for task input/output

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: protos/flyteidl/service/dataproxy.proto [hunk 1]
```diff
@@ -90,6 +94,9 @@ message CreateDownloadLinkRequest {
     // NodeId is the unique identifier for the node execution. For a task node, this will retrieve the output of the
     // most recent attempt of the task.
     core.NodeExecutionIdentifier node_execution_id = 3;
+
+    // can add a flyte url as an option for the source
+    string flyte_url = 4;
```
Comment (kumare3) [inline protos/flyteidl/service/dataproxy.proto:99]: why is this here?
